enhancements/two-nodes-openshift/2no_template.md

```diff
@@ -0,0 +1,589 @@
+   We should be safe after [MGMT-13586](https://issues.redhat.com/browse/MGMT-13586) which makes the installer wait for the bootstrap etcd member to be removed first before shutting it down.
+3. After the bootstrapped node is removed add it to the cluster as a “regular” node.
+4. Switch CEO to “2NO” mode (where it does not manage etcd) and remove the etcd static pods
+   * [TBD localized/global switch]
+   * This is done because we want to allow simpler maintenance and keeping  some of CEO functionality (defragmentation, cert rotation ect…)
+5. Configure Pacemaker/Corosync to manage etcd/kubelet/cri-o
+6. [TBD storage]
+
+#### Fencing / quorum management
+Fencing quorum is managed by corosync and etcd  will be managed by Pacemaker which will force the etcd quorum.
+
+Here is a node failure example demonstrating that:
+1. Corosync on the survivor (Node1)
+2. Etcd loses internal quorum (E-quorum) and goes read-only
+3. Node1 retains “corosync quorum” (C-quorum) and initiates fencing of Node2
+4. Once fencing is successful Pacemaker will use a fence/resource agent (TBD) which will reschedule the workload from the fenced node
+5. Pacemaker on Node1 forces E-quorum (etcd promotion event)
+6. Cluster continues with no redundancy
+7. … time passes …
+8. Node2 has a persistent failure that prevents communication with Node1
+   * Node2 does not have C-quorum (requires forming a membership with it’s peer)
+   * Node2 does not start etcd or kubelet, remains inert waiting for Node1
+9. Persistent failure on Node2 is repaired
+10. Corosync membership containing both nodes forms
+11. Pacemaker “starts” etcd on Node2 as a follower of Node1
+12. Pacemaker “promotes” etcd on Node2 as full replica of Node1
+13. Pacemaker starts kubelet
+14. Cluster continues with 1+1 redundancy
+
+[Here](#failure-handling) is a  more extensive list of failure scenarios.
+
+#### CEO Enhancement
+1. Requires a new infrastructure type in OpenShift APIs
+2. Replicate the functionality of CEO that is tied to static pods? e.g. Certificate rotation, backup/restore, apiserver etcd endpoint controller
+3. Some functionality to “know” when to switch to “2NO mode”
+
+### Risks and Mitigations
+
+#### Risks:
+
+1. On a failed node scenario ,The node with elected Pacemaker DC (designated coordinator) will fence the other node. That brings up the question of how to schedule the workload of the fenced nodes ? [mitigation](#scheduling-workload-on-fenced-nodes)
+2. We plan to configure Pacemaker to manage etcd and give it quorum (for example after the remaining node fence its peer in a failed node use case)  [mitigation](#pacemaker-controlling-key-elements)
+   1. How do we plan Pacemaker to give the etcd quorum and which consideration should be taken ?
+   2. How does Pacemaker giving etcd quorum affect other etcd stakeholders (etcd pod, etcd operator, etc…)  ?
```

Comment:
What's the risk here?  An open question is different from a risk.

Instead of "How do we do X?" type questions, try to think in terms of "If we do (or can't do) X then {risk}".

Reply:
And the open questions, should be listed in the section at line ~388

Reply:
Resolved